aptos-move/move-examples/post_mint_reveal_nft/sources/whitelist.move

```diff
@@ -0,0 +1,145 @@
+        whitelist_mint_price: u64,
+        whitelist_minting_start_time: u64,
+        whitelist_minting_end_time: u64,
+    }
+
+    /// The whitelist start time is not strictly smaller than the whitelist end time.
+    const EINVALID_WHITELIST_SETTING: u64 = 1;
+    /// The whitelist stage should be added in order. If the whitelist_stage parameter is not equal to the length of the whitelist_configs vector,
+    /// it means that the whitelist stage is not added in order and we need to abort.
+    const EINVALID_STAGE: u64 = 2;
+    /// The specified address does not exist.
+    const EACCOUNT_DOES_NOT_EXIST: u64 = 3;
+    /// The amount that the user wants to mint exceeds the amount that the user is allowed to mint.
+    const EEXCEEDS_MINT_LIMIT: u64 = 4;
+    /// Cannot add more addresses to the whitelist after this whitelist minting stage already ends.
+    const EINVALID_UPDATE_AFTER_MINTING: u64 = 5;
+    /// The given address is not on the whitelist.
+    const EACCOUNT_NOT_WHITELISTED: u64 = 6;
+
+    #[view]
+    /// Checks if WhitelistMintConfig resource exists.
+    public fun whitelist_config_exists(module_address: address): bool {
+        exists<WhitelistMintConfig>(module_address)
+    }
+
+    #[view]
+    /// Checks if the current time is within a whitelist stage. Return value bool is true if the current time is within a whitelist stage.
+    /// Return value u64 is the index of the whitelist stage that the current time is within.
+    public fun get_whitelist_stage_by_current_time(module_address: address): (bool, u64) acquires WhitelistMintConfig {
+        // If the project doesn't have a whitelist, return false.
+        if (!whitelist_config_exists(module_address)) {
+            return (false, 0)
+        };
+        let whitelist_mint_config = borrow_global<WhitelistMintConfig>(module_address);
+        let now = timestamp::now_seconds();
+
+        let i = 0;
+        while (i < vector::length(&whitelist_mint_config.whitelist_configs)) {
+            let whitelist_stage = vector::borrow(&whitelist_mint_config.whitelist_configs, i);
+            if (whitelist_stage.whitelist_minting_start_time <= now && now < whitelist_stage.whitelist_minting_end_time) {
+                return (true, i)
+            };
+            i = i + 1;
+        };
+        return (false, 0)
+    }
+
+    #[view]
+    /// Returns the number of total stages available.
+    public fun get_num_of_stages(module_address: address): u64 acquires WhitelistMintConfig {
+        vector::length(&borrow_global<WhitelistMintConfig>(module_address).whitelist_configs)
+    }
+
+    #[view]
+    /// Checks if the given address is whitelisted in the specified whitelist stage.
+    /// Return value u64 is the minting price of the current stage.
+    /// Return value bool is true if the given address is whitelisted.
+    public fun is_user_eligible_for_minting(module_address: address, minter_address: address, stage: u64): (u64, bool) acquires WhitelistMintConfig {
```

Comment:
Why not check against any current active whitelist mint stage? May be we can make a separate function that says is_user_currently_eligible_for_whitelisted_minting. This can save the client from having to send 2 requests, one to get current stage and one to see if the user is eligible.